### ISSUE TITLE
Use mmap for communication with IfcEngine

### ISSUE THREAD

**Issue by aothms** [opensourceBIM/BIMserver#293]
I have a ticket here https://github.com/IfcOpenShell/IfcOpenShell/issues/22 that proposes the use of mmap'ed files so that the IFC data can be shared by the BIMserver and IfcEngine process. By removing this duplication some of the memory usage during the early checkin phase will be removed and the process can start sooner because the file contents are read lazily. Not sure how it will impact the total memory footprint as I assume the IFC data in Java would already have been freed when the IfcEngine is still processing (and hence allocating more memory). The ticket also contains Java code to set this up. Do you want to give it a try?

**Comment by rubendel** [comment]
This might be interesting, although in BIMserver the IFC files are created in a streaming matter (the files a user is checking in are not the files that are being sent to the ifc engine), so no memory to win on that side.

The reasons BIMserver is not sending through the original files:
- Not all objects are relevant for geometry, those are excluded (the biggest part is still geometry though)
- Some (actually a lot of) files are too big to process in one go, so we split them
- By splitting them we also get some sort of crude progress-indication
- We can utilize multiple CPU cores

So in short, great contribution, but I don't think it's useful to implement for BIMserver at the moment.

My feeling is that the generated geometry usually takes up at least 10 times the amount of memory the IFC model takes, it would be nice to have a way to get the instance data per instance in a way that IOS can throw away the previous one, but that's probably not how it's working at the moment?
